my-app/src/Components/Counter.jsx

```diff
@@ -0,0 +1,32 @@
+import React from "react";
+
+class Counter extends React.PureComponent {
+  constructor() {
+    super();
+    this.increase = this.increase.bind(this);
+  }
+
+  state = {
+    counter: 0,
+  };
+
+  increase() {
+    this.setState({ counter: this.state.counter + 1 });
```

Comment:
React may update state asynchronously, so if you rely on previous state, you should use function in `setState`:
```js
increase = () => {
  this.setState(prevState => ({ counter: prevState.counter + 1 }));
}
```